vision_evaluation/evaluators.py

```diff
@@ -1046,3 +1046,78 @@ def add_predictions(self, predictions, targets):
             gt_mask = np.asarray(gt_mask)
             mean_l1 = np.abs(pred_mask.astype(np.float)-gt_mask.astype(np.float)).mean()
             self._metric_sum += mean_l1
+
+
+class GroupWiseEvaluator(Evaluator):
+    """
+    Group wise evaluator: calculates metrics for the provided evaluator for each target class and group class separately.
+    """
+
+    def __init__(self, evaluator: Evaluator):
+        super().__init__()
+        self._evaluator = evaluator
+        self.store = None
+
+    def _get_id(self):
+        return "group_wise_accuracy"
+
+    def _store_by_group(self):
+        """
+            store dict will save target class labels and predictions corresponding to each group class label.
+
+            store = {(target_class, group_class) : {
+                "predictions":[]
+                "targets":[]
+            }}
+            target_class: 0,1,2...
+            group_class: 0,1,2...
+        """
+        for (target_class, group_class), prediction in zip(self.ground_truths, self.predictions):
+            self.store[(target_class, group_class)]['predictions'].append(prediction)
+            self.store[(target_class, group_class)]['targets'].append(target_class)
+
+    def add_predictions(self, predictions, ground_truths):
+        """ Evaluate a batch of predictions.
+        Args:
+            predictions: the model output numpy array. Shape (N, num_class)
+            ground_truths: the golden truths numpy array. Shape (N,2). The first column is the target class label and the second column is the group class label.
+        """
+        assert len(predictions) == len(ground_truths)
+        self.ground_truths = ground_truths
+        self.predictions = predictions
+
+        if not self.store:
+            self.store = collections.defaultdict(
+                lambda: collections.defaultdict(list))
+
+        for (target_task, group), prediction in zip(self.targets, self.predictions):
+            self.store[(target_task, group)]['predictions'].append(prediction)
+            self.store[(target_task, group)]['targets'].append(target_task)
+
+    def add_predictions(self, predictions, targets):
```

Comment:
There are two add_predictions() methdo now.

Reply:
my bad. something got messed up during merge conflict resolution. hopefully should be fixed now

Reply:
updated